### DISCUSSION TITLE
How to add CSS modules Go to definition support?

### DISCUSSION THREAD

**Discussion by thezaff** [LazyVim/LazyVim#4074] [Q&A]
What I already tried, adding a plugin cssmodules.lua:
```
return {
	"neovim/nvim-lspconfig",
	opts = function(_, opts)
		LazyVim.extend(opts.servers.vtsls, "settings.vtsls.tsserver.globalPlugins", {
			{
				name = "typescript-plugin-css-modules",
				location = "$env.FNM_DIR/aliases/default/lib/node_modules/typescript-plugin-css-modules",
				enableForWorkspaceTypeScriptVersions = true,
			},
		})
	end,
}
```
And it doesn't work.

**Comment by beetol** [comment]
Same issue here. When my cursor is like `import st|yle from './styles.module.scss`, it takes me to the global.d.ts for css modules.

Personally, this is making me not want to use Neovim. Neovim is advertised as an editor that gives you superpowers. If the lsp can't handle goto for css/scss files, then it's actually a dangerous editor for a professional programmer (it'll get them fired for losing sleep trying to make this work, hence poor performance at work).

Webstorm handles this with poise when you put the cursor as shown above. Takes you right to the file you want. It's quite a shame there is ZERO discussion on this as little things like that can waste a whole week trying to tweak your config.

**Reply by jsantos42** [reply]
Any update here? It's a PINA to not have goToDefinition in css/sass/less

**Reply by thezaff** [reply]
It worked for me in July 2024 with following config. I am not using neovim anymore, just found it from history of a chat where I tried to fix it with help of community. In my message from that chat, I said (no idea why): 

> I needed to pass default argument to lsp config, here is code
```
return {
 "neovim/nvim-lspconfig",
 opts = {
  servers = {
   cssmodules_ls = {
    init_options = { camelCase = "dashes" },
   },
  },
}
```
